### DISCUSSION TITLE
Unrecognized option '-canny'

### DISCUSSION THREAD

**Discussion by AndyBern1** [ImageMagick/ImageMagick#7393] [Help]
### ImageMagick version

6.9.11-60

### Operating system, version and so on

Linux Mint 21.2 MATE

### Description

I'm trying to clean up a bunch of scanned sheet music pages, and it errors on this command:

`convert image1.png -canny 0x1+10%+30% -colorspace gray -blur 0x20 -equalize image2.png`

The error message is:

`convert-im6.q16: unrecognized option '-canny 0x1+10%+30%' @ error/convert.c/ConvertImageCommand/1166.`

I also tried version 7.1.1.33 (after uninstalling the previous version) and it doesn't recognize "canny" either. Why is "canny" not recognized? I couldn't find anything online about this.

**Comment by fmw42** [comment]
That should work, but I do not understand why you would use -canny and a large blur on sheet music to clean it up?  Nevertheless, I can only think of 3 issues:  1) bad image and 2) buggy versions of Imagemagick and 3) bad installs of imagemagick.

With IM 7, you should use magick, not convert.  Does that work with magick?

Post your image so we can test with it.

**Reply by AndyBern1** [reply]
Backtracking a bit, I was using smartcrop in my script and, due to an issue that generates an error if the image is smaller than the crop dimensions, I was troubleshooting by replacing the full smartcrop script with just the relevant commands, and changing the tmpI1.mpc and tmpI2.mpc files to pngs to see what was going on.

Anyway, I just discovered the command works when entered directly, so it's probably something with my script. I'm gonna work on it some more. It's just weird that the problem would manifest itself as canny not being unrecognized.

In version 7, I used 'magick' instead of 'convert', and it is a normal space after -canny.

**Reply by fmw42** [reply]
What is smartcrop and where did you get it? Is that one of my scripts and do you have a license to use it if for commercial purposes?

**Reply by AndyBern1** [reply]
It's one of your scripts, and it's personal use.

**Reply by AndyBern1** [reply]
I have an old music book that's falling apart and not available anymore. I wanted to save it by scanning  and converting it into a pdf. I can do so manually with GIMP, but I figure scripting is a good way to help me learn imagemagick.

**Reply by fmw42** [reply]
OK. No problem. Thanks for the clarification.

**Comment by snibgo** [comment]
> convert-im6.q16: unrecognized option '-canny 0x1+10%+30%' @ error/convert.c/ConvertImageCommand/1166.

For come reason, IM is reading the option as `-canny 0x1+10%+30%`. IM should be splitting the command at the space, but it isn't. If I use an option that doesn't exist, like `-cannyXX`, I get this:
```
D:\temp>%IM%convert toes.png -cannyXX 0x1+10%+30% info:
convert.exe: unrecognized option `-cannyXX' @ error/convert.c/ConvertImageCommand/1166.
```
Note that the error message does not contain `0x1+10%+30%`.

So check that your space after `-option` is an ordinary space character.

**Comment by fmw42** [comment]
I copied and paste your code and ran it with one of my images and it worked fine on IM 7.1.1.33 Mac OSX.
